File: Tutorial5-3Dagger-Android-MutipleActivitiesAndFragments/src/main/java/com/example/tutorial5_3dagger_android_mutipleactivitiesandfragments/MyFragmentModule.java
package com.example.tutorial5_3dagger_android_mutipleactivitiesandfragments;


import android.content.Context;
import android.content.SharedPreferences;

import com.example.tutorial5_3dagger_android_mutipleactivitiesandfragments.MainActivity;
import com.example.tutorial5_3dagger_android_mutipleactivitiesandfragments.MyFragment;
import com.example.tutorial5_3dagger_android_mutipleactivitiesandfragments.di.FragmentScope;
import com.example.tutorial5_3dagger_android_mutipleactivitiesandfragments.model.MySharedPreferences;
import com.example.tutorial5_3dagger_android_mutipleactivitiesandfragments.model.ToastMaker;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;

@Module
public abstract class MyFragmentModule {

    @FragmentScope
    static MySharedPreferences provideMySharedPreference(SharedPreferences sharedPreferences) {
        return  new MySharedPreferences(sharedPreferences);
    }

}

File: Tutorial5-4Dagger-Android-MultipleActivitiesAndFragments/src/main/java/com/smarttoolfactory/tutorial5_4dagger_android_multipleactivitiesandfragments/main/MyFragmentModule.kt
package com.smarttoolfactory.tutorial5_4dagger_android_multipleactivitiesandfragments.main

import android.content.SharedPreferences
import com.example.tutorial5_3dagger_android_multipleactivitiesandfragments.di.scope.FragmentScope
import com.smarttoolfactory.tutorial5_4dagger_android_multipleactivitiesandfragments.model.MySharedPreferences
import dagger.Module
import dagger.Provides


@Module
object MyFragmentModule {

    @JvmStatic
    @FragmentScope
    @Provides
    fun provideMySharedPreference(sharedPreferences: SharedPreferences): MySharedPreferences {
        return MySharedPreferences(sharedPreferences)
    }

}